src/arithmetic.cpp

```diff
@@ -0,0 +1,330 @@
+// ���������� ������� � ������� ��� ���������� �������������� ���������
+
+#include "arithmetic.h"
+#include <string>
+#include "stack.h"
+#include "poly.h"
+using namespace std;
+ArithmeticExpr::ArithmeticExpr(string infx) {
+	infix = infx;
+}
+int StringIsAlmostCorrect(string s) {
+	int count = 0;
+	for (int i = 0; i < s.length(); i++) {
+		if ((s[i] == '(' || s[i] == '+' || s[i] == '-' || s[i] == '*' || s[i] == '/') && (s[i + 1] == ')' || s[i + 1] == '+' || s[i + 1] == '*' || s[i + 1] == '/')) {
+			return i; //checking for errors (*, +*, etc.
+		}
+		if (s[i] == ')' && s[i + 1] == '(') return i; // checking for )(
+		if (s[i] == ' ') return i; // no gaps
+		if (s[i] == '(') count++;
+		if (s[i] == ')') count--;
+		if (i >= 1) {
+			if (s[i] == 'x' && (s[i - 1] == '0' || s[i - 1] == '1' || s[i - 1] == '2' || s[i - 1] == '3' || s[i - 1] == '4' || s[i - 1] == '5' || s[i - 1] == '6' || s[i - 1] == '7' || s[i - 1] == '8' || s[i - 1] == '9')) {
```

Comment:
```suggestion
		if (i >= 1) {
			if (s[i] == 'x' && s[i - 1] >= '0' && s[i - 1] <= '9') {
```
так компактнее будет